### PR TITLE
Make APP_PORT optional

### PR DESCRIPTION
**Pull request by rphcrosbyhf**
- Changes container port configuration to make port optional

### CODE REVIEW

**Comment by rphcrosbyhf** [comment]
@ahelal you'll have to excuse my lack of knowledge on this. Is there anything else that needs changing here?

**Comment by ahelal** [comment]
@rphcrosbyhf on top of my head that looks right, but we need to add a test for it.

@michael-robertson can we use hooks for tests. hooks does not need to announce a service?

File: README.md [hunk 1]
```diff
@@ -250,7 +250,7 @@ ENV APP_NAME_1        "pilot-multiapp-local"
 ENV APP_PORT_1        8890
 ENV APP_TAGS_1        "TEST,v1.1.1,Development,local"
 ```
-Caveat: the `APP_PORT_$NUM` is *not* required. If omitted, service is not registered in consul. 
+Caveat: the `APP_PORT_$NUM` is *not* required. If omitted, service is not registered in consul.
```
Comment (aleerizw) [inline README.md]: Maybe remove the caveat since the behavior is consistent now?

File: README.md [hunk 2]
```diff
@@ -235,7 +235,7 @@ The group of variables that define application configuration.
 |-------------------|----------|---------|-------------|
 |APP_NAME           |  YES     |         | Application name|
 |APP_EXEC           |  YES     |         | Application execution command|
-|APP_PORT           |  YES     |         | Application Port |
+|APP_PORT           |  NO      |         | Application Port |
```
Comment (aleerizw) [inline README.md]: Maybe document the behavior when the variable is not defined here?

**Comment by aleerizw** [comment]
+1

**Comment by michael-robertson** [comment]
+1

**Pull request merged**